src/quacc/runners/prep.py

```
@@ -174,3 +177,25 @@ def terminate(tmpdir: Path | str, exception: Exception) -> None:
         symlink_path.symlink_to(job_failed_dir, target_is_directory=True)
 
     raise JobFailure(job_failed_dir, message=msg, parent_error=exception) from exception
+
+
+def rmtree(*args, max_retries=3, retry_wait=10, **kwargs):
+    """
+    rmtree that will retry if we get common NFS errors (files still being deleted, etc)
+    Adapted from https://github.com/teojgo/reframe/blob/master/reframe/utility/osext.py
+    """
+    if "onerror" in kwargs and kwargs["onerror"] is not None:
+        shutil.rmtree(*args, **kwargs)
+        return
+
+    for i in range(max_retries):
+        try:
+            shutil.rmtree(*args, **kwargs)
+            return
+        except OSError as e:
+            if i == max_retries:
+                raise
```

Could we raise a specific exception and have a useful message?